mesonbuild/interpreter.py

```diff
@@ -1615,10 +1616,15 @@ def has_multi_arguments_method(self, args, kwargs):
     @permittedKwargs({})
     def get_supported_arguments_method(self, args, kwargs):
         args = mesonlib.stringlistify(args)
-        supported_args = []
-        for arg in args:
-            if self.has_argument_method(arg, kwargs):
-                supported_args.append(arg)
+        with conc.ThreadPoolExecutor() as executor:
```

Comment:
concurrent.futures has a lot of overhead, and it's not really needed either. We're just running processes and waiting for them to exit. We already benchmarked all the various methods in https://github.com/mesonbuild/meson/issues/3635 and `subprocess.Popen` is twice as fast as this or async/await.

Reply:
Also the same approach was discussed in https://github.com/mesonbuild/meson/pull/4904. I really think we should reorganize the code to allow using Popen instead of trying to keep the changes minimal and picking a parallelization method with a lot of overhead.

Reply:
Totally forgot about those 2, thanks! I noticed that gstreamer has many places where it's not using get_supported_args(), need to fix that first to get proper benchmark numbers.